CMakeLists.txt

```diff
@@ -0,0 +1,96 @@
+cmake_minimum_required(VERSION 3.15)
```

Comment:
Question: I don't see this cmake file used elsewhere, is it necessary to be included here? I don't have strong feelings against cmake, it's just that I feel we should stay consistent with one: if the current makefile is enough, maybe we can remove this one?

Reply:
It's fine to remove it. It would be great if we can still keep simulator for early phase development(e.g. keep simulator folder and MACRO USE_SIM).  Also need to add a Makefile in simulator folder.

Reply:
I think we can definitely keep the simulator, and if you want to use CMake for simulator, we can do that, but maybe moving this file to the simulator directory instead? Or the simulator, is better maintained as a standalone project, thoughts?

Reply:
OK. Keep the simulator as a standalone project under "simulator" folder and change CMake to Makefile for simplicity.